common/StringUtil.cpp

```diff
@@ -206,6 +207,22 @@ namespace StringUtil
 		return ss.str();
 	}
 
+	std::string toLower(const std::string& input)
+	{
+		std::string newStr;
+		std::transform(input.begin(), input.end(), std::back_inserter(newStr),
+			[](unsigned char c) { return std::tolower(c); });
+		return newStr;
+	}
+
+	bool compareNoCase(const std::string& str1, const std::string& str2)
+	{
+		return std::equal(str1.begin(), str1.end(), str2.begin(),
+			[](char a, char b) {
+				return tolower(a) == tolower(b);
+			});
```

Comment:
This is almost certainly unsafe as access to `str2` is unbounded. You should return false before that line if string lengths are not equal.

Reply:
Also redundant since `Strcasecmp` already exists, although a string_view overload could be nice.